routes/bitcoin.go

```diff
@@ -0,0 +1,219 @@
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+
+	"../db/models"
+	"../helpers"
+	"github.com/gorilla/mux"
+)
+
+// BitcoinData -> this structure is the base response for the project for bitcoins
+type BitcoinData struct {
+	Success bool             `json:"success"`
+	Data    []models.Bitcoin `json:"data"`
+	Errors  []string         `json:"errors"`
+}
+
+// ResponseData -> this structure is used to adapt to the response sent by the coinbase api
+type ResponseData struct {
+	Data models.Bitcoin `json:"data"`
+}
+
+// CreateBitcoin -> Manually does a entry in the db
+func CreateBitcoin(w http.ResponseWriter, req *http.Request) {
+	bodyBitcoin, success := helpers.DecodeBody(req)
+	if success != true {
+		http.Error(w, "could not decode body", http.StatusBadRequest)
+		return
+	}
+
+	var data = BitcoinData{Errors: make([]string, 0)}
+	if !helpers.IsValidBase(bodyBitcoin.Base) {
+		data.Success = false
+		data.Errors = append(data.Errors, "invalid base")
+
+		json, _ := json.Marshal(data)
+
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		w.Write(json)
+		return
+	}
+
+	btc, success := models.InsertBitcoinValue(bodyBitcoin.Base, bodyBitcoin.Currency, bodyBitcoin.Amount)
+	if success != true {
+		data.Errors = append(data.Errors, "could not create Entry")
+	}
+
+	data.Success = success
+	data.Data = append(data.Data, btc)
+
+	json, _ := json.Marshal(data)
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	w.Write(json)
+	return
+}
+
+// GetBicoins -> gets a list of al bitcoins
+func GetBicoins(w http.ResponseWriter, req *http.Request) {
+	var btcs []models.Bitcoin = models.GetAllValues()
+
+	var data = BitcoinData{true, btcs, make([]string, 0)}
+	json, _ := json.Marshal(data)
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	w.Write(json)
+}
+
+// GetBicoin -> gets a list of al bitcoins
+func GetBicoin(w http.ResponseWriter, req *http.Request) {
+	var btcs []models.Bitcoin
+	var data = BitcoinData{true, btcs, make([]string, 0)}
+	vars := mux.Vars(req)
+
+	ID, err := strconv.Atoi(vars["id"])
+	if err != nil {
+		data.Success = false
+		data.Errors = append(data.Errors, "Invalid ID")
+		json, _ := json.Marshal(data)
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write(json)
+		return
+	}
+	btc, success := models.GetValue(ID)
+	if success != true {
+		data.Errors = append(data.Errors, "Could not retrive information from db")
+		json, _ := json.Marshal(data)
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusBadRequest)
+		w.Write(json)
+		return
+	}
+
+	data.Data = append(data.Data, btc)
+
+	json, _ := json.Marshal(data)
+
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	w.Write(json)
+}
+
+// GetBitcoinFromAPI -> gets the bitcoin data from coinbase
+func GetBitcoinFromAPI(w http.ResponseWriter, r *http.Request) {
```

Comment:
I feel like we should also send/save the info we get from the API rather than just printing it.

Reply:
This was just to retrive the info, on the post `/bitcoin` is where I save the data